tests/test_assembly.py

```diff
@@ -168,28 +184,41 @@ def test_toJSON(simple_assy, nested_assy, empty_top_assy):
     assert len(r3) == 1
 
 
-def test_save(simple_assy, nested_assy):
+def test_save(nested_assy, nested_assy_sphere):
+
+    nested_assy.save("nested.step")
+    assert os.path.exists("nested.step")
+
+    nested_assy.save("nested.xml")
+    assert os.path.exists("nested.xml")
+
+    nested_assy.save("nested.step")
+    assert os.path.exists("nested.step")
+
+    nested_assy.save("nested.stp", "STEP")
+    assert os.path.exists("nested.stp")
+
+    nested_assy.save("nested.caf", "XML")
+    assert os.path.exists("nested.caf")
```

Comment:
Are you interested in converting this to a pytest parameterized test? If so just let me know and I'll push the following (GitHub won't let me make suggestions over deleted lines)
```python
@pytest.mark.parametrize(
    "extension, args",
    [
        ("step", ()),
        ("xml", ()),
        ("stp", ("STEP",)),
        ("caf", ("XML",)),
        ("wrl", ("VRML",)),
    ],
)
def test_save(extension, args, nested_assy, nested_assy_sphere):

    filename = "nested." + extension
    nested_assy.save(filename, *args)
    assert os.path.exists(filename)


def test_save_gltf(nested_assy_sphere):

    nested_assy_sphere.save("nested.glb", "GLTF")
    assert os.path.exists("nested.glb")
    assert os.path.getsize("nested.glb") > 50 * 1024


def test_save_vtkjs(nested_assy):

    nested_assy.save("nested", "VTKJS")
    assert os.path.exists("nested.zip")
```

Parameterize would prevent problem from a previous comment, where the first save changed the state of the assembly and affected testing a following save.

If you're not interested note that `nested_assy.save("nested.step")` currently appears twice.

Reply:
Good idea, please do push!